liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -0,0 +1,889 @@
+                    elements.add(element(name("configFile"), configFile.getCanonicalPath()));
+                } else if (goal.equals("create-server")) {
+                    elements.add(element(name("configFile"), configFile.getCanonicalPath()));
+                    if (assemblyArtifact != null) {
+                        Element[] featureElems = new Element[4];
+                        featureElems[0] = element(name("groupId"), assemblyArtifact.getGroupId());
+                        featureElems[1] = element(name("artifactId"), assemblyArtifact.getArtifactId());
+                        featureElems[2] = element(name("version"), assemblyArtifact.getVersion());
+                        featureElems[3] = element(name("type"), assemblyArtifact.getType());
+                        elements.add(element(name("assemblyArtifact"), featureElems));
+                    }
+                }
+            }
+        } catch (IOException e) {
+            log.error("Unable to resolve canonical paths " + e.getMessage());
+        }
+        return elements.toArray(new Element[elements.size()]);
+    }
+
+    private void runMojo(String plugin, String goal, String serverName, List<String> dependencies)
+            throws MojoExecutionException {
+        Plugin mavenPlugin = project.getPlugin(plugin);
+        log.info("plugin version: " + mavenPlugin.getVersion());
+        executeMojo(mavenPlugin, goal(goal),
+                configuration(getPluginConfigurationElements(goal, serverName, dependencies)),
+                executionEnvironment(project, session, pluginManager));
+    }
+
+    private static MavenProject loadProject(File pomFile) throws IOException, XmlPullParserException {
+        MavenProject ret = null;
+        MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+
+        if (pomFile != null && pomFile.exists()) {
+            FileReader reader = null;
+            try {
+                reader = new FileReader(pomFile);
+                Model model = mavenReader.read(reader);
+                model.setPomFile(pomFile);
+                ret = new MavenProject(model);
+            } finally {
+                reader.close();
+            }
+        }
+        return ret;
+    }
+
+    private void listFiles(File directory, List<File> files, String suffix) {
+        if (directory != null) {
+            // Get all files from a directory.
+            File[] fList = directory.listFiles();
+            if (fList != null) {
+                for (File file : fList) {
+                    if (file.isFile() && ((suffix == null) || (file.getName().toLowerCase().endsWith("." + suffix)))) {
+                        files.add(file);
+                    } else if (file.isDirectory()) {
+                        listFiles(file, files, suffix);
+                    }
+                }
+            }
+        }
+    }
+
+    private class ServerFeature extends ServerFeatureUtil {
```

Comment:
This seems like it should just be an implementation of a common logger as I discussed in the ci.common review. Also I noticed this is being instantiated multiple times in this class, this should be able to be avoided if we make it a singleton.

Reply:
Logger interface will be tracked separately in https://github.com/WASdev/ci.common/issues/51

Changed current usage into a single instance with https://github.com/WASdev/ci.maven/pull/476/commits/7ca73f1f75416c40322aa30c781fe46343bd17d1 (can't use proper singleton convention since this is a nested class, but functionally it is the same)